pkg/kubeapiserver/options/authentication.go

```diff
@@ -584,3 +635,59 @@ func (o *BuiltInAuthenticationOptions) ApplyAuthorization(authorization *BuiltIn
 		o.Anonymous.Allow = false
 	}
 }
+
+func (o *BuiltInAuthenticationOptions) validateOIDCOptions() []error {
+	var allErrors []error
+
+	// Existing validation when jwt authenticator is configured with oidc-* flags
+	if len(o.AuthenticationConfigFile) == 0 {
+		if o.OIDC != nil && areOIDCFlagsConfigured() && (len(o.OIDC.IssuerURL) == 0 || len(o.OIDC.ClientID) == 0) {
```

Comment:
Why are the issuer URL and client ID check not required when the the AuthenticationConfigFile is specified?

Reply:
found them in ValidateAuthenticationConfiguration .